coltypes/coltypes.go

```diff
@@ -52,24 +54,32 @@ func TransformRow(row map[string]any, columnTypes map[string]ColumnData) (map[st
 			continue
 		}
 
-		switch data := columnTypes[key]; {
-		case data.Type == numberType:
-			if data.Precision != nil && *data.Precision == 1 && data.Scale != nil && *data.Scale == 0 {
-				if v, ok := value.(int64); ok {
-					if v == 1 {
-						result[key] = true
-					} else {
-						result[key] = false
-					}
-
-					continue
-				}
+		if columnTypes[key].Type == numberType {
+			valStr, err := godror.Num.ConvertValue(value)
+			if err != nil {
+				return nil, fmt.Errorf("convert oracle number type to string: %w", err)
+			}
+
+			value, err = strconv.Atoi(valStr.(string))
```

Comment:
if valStr has string type, maybe we don't need convert it

Reply:
It's an interface.